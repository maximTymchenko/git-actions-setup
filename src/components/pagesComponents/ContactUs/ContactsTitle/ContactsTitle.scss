@import '../../../../styles/vars';

.contact-us-page__info {
    max-width: 600px;
    font-family: 'OktaNeue-Regular';
    font-size: 22px;
    line-height: 1.5;
    color: $text__body;

    @media screen and (max-width: $mobile) {
        font-size: 18px;
    }
}
