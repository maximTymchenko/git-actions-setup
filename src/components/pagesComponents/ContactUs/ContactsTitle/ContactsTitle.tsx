import React, { FC } from 'react';
import MainTitle from '../../../common/MainTitle';
import './ContactsTitle.scss';

interface ITitleProps {
    subtype: string;
    type: string;
    subtitle: string;
    title: string;
    titleText: string;
    subtitleText: string;
    titleType: string;
    titleSubtype: string;
}

const ContactsTitle: FC = (props: ITitleProps) => {
    return (
        <div data-testid="contacts-title" className="contact-us-page__headline">
            <div className="contact-us-page__wrapper">
                <MainTitle
                    subtype={props.titleSubtype}
                    type={props.titleType}
                    subtitleF={props.subtitleText}
                    title={props.titleText}
                />
            </div>

            <p className="contact-us-page__info">{props.text}</p>
        </div>
    );
};

export default ContactsTitle;
