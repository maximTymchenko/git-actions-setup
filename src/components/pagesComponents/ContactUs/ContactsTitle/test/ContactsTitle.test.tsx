import React from 'react'
import ContactsTitle from '../ContactsTitle'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <ContactsTitle subtitleText="test" titleText="test"></ContactsTitle>
  )
  expect(getByTestId('contacts-title')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(
      <ContactsTitle subtitleText="test" titleText="test"></ContactsTitle>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
