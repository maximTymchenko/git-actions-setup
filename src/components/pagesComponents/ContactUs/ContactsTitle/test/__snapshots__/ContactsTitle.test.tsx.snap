// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="contact-us-page__headline"
  data-testid="contacts-title"
>
  <div
    className="contact-us-page__wrapper"
  >
    <div>
      <div
        className="title__wrapper"
      >
        <h2
          className="title title--small"
          data-testid="section-title"
        >
          test
        </h2>
        <h2
          className="title title--bold"
          data-testid="section-title"
        />
        <h2
          className="title title--small"
          data-testid="section-title"
        />
      </div>
      <h1
        className="title title--undefined"
        data-testid="main-title"
      >
        test
      </h1>
    </div>
  </div>
  <p
    className="contact-us-page__info"
  />
</div>
`;
